component/common/loki/types.go

```diff
@@ -147,3 +147,55 @@ func AddLabelsMiddleware(additionalLabels model.LabelSet) EntryMiddleware {
 		})
 	})
 }
+
+// LogsFanout is an abstraction for fanning out log entries to multiple
+// receivers.
+type LogsFanout struct {
+	mut       sync.RWMutex
+	receivers []LogsReceiver
+}
+
+// NewFanout creates new LogsFanout.
+func NewFanout(r []LogsReceiver) *LogsFanout {
```

Comment:
Should we go one layer up, and include the handler which receives the entries within the Fanout itself, instead of having to receive it and call the Send method every time from the component's Run method?

LogsFanout could be started in its own rungroup and pass along entries in the background.

Reply:
I would be in favour of including the logic that reads from a channel, as it will further reduce duplication. Could be part of `LogFanout`, but could also be a new struct that takes `LogFanout` in constructor.